tests/core/test_retrograde_periods.py

```
@@ -0,0 +1,135 @@
+            assert [(p.start_clipped, p.end_clipped) for p in res.periods] == [(True, False)], sd.julian_day
+            assert res.periods[0].end_jd == sd.julian_day
+
+    def test_a_bound_half_a_second_off_the_station_keeps_the_real_station(self):
+        # Half a second is far beyond the solver's resolution: the station is
+        # then a real station inside the range, or simply outside it — never a
+        # bound. Mercury, June-July 2020 (SR 18 June, SD 12 July).
+        stations = RetrogradeStationFactory.from_iso_range("2020-06-01", "2020-07-31", ["Mercury"]).stations
+        sr = next(s for s in stations if s.station_type == "SR")
+        sd = next(s for s in stations if s.station_type == "SD")
+        half = 0.5 / 86400.0
+        periods = lambda a, b: RetrogradeStationFactory.retrograde_periods_from_julian_day(a, b, ["Mercury"]).periods
```

Ruff (0.16.2)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `periods` as a `def`

(E731)